### PR TITLE
Move Tags/Controller over to a real Controller

### PR DESCRIPTION
**Pull request by rullzer**
Follow up of https://github.com/nextcloud/server/pull/947

This moves the tags 'controller' over to a real controller and leverages all the DI the AppFramework gives us.

There were no tests. And are no tests so far. If somebody desperatly wants them feel free to add.

CC: @nickvergessen @MorrisJobke @LukasReschke @icewind1991 

P.s. I still vote to kill this off in time.

### CODE REVIEW

**Comment by rullzer** [comment]
Lets deprecate this internal route (How do we signal this?)
And remove this in NC12. As I dont' believe it is used anywhere.

**Comment by nickvergessen** [comment]
Announce in the dev forum

File: core/Controller/TagsController.php [hunk 1]
```diff
@@ -0,0 +1,232 @@
+<?php
+/**
+ *
```
Comment (LukasReschke) [inline core/Controller/TagsController.php]: Missing @copyright header?

Comment (rullzer) [inline core/Controller/TagsController.php]: Fair enough let me copy

**Comment by LukasReschke** [comment]
LGTM

**Comment by rullzer** [comment]
So it seems in the old code the route

/tags/files/ids

Throws an exception.
It becomes less and less likely any of this stuff is used.

I vote again to remove.

**Pull request closed without merging**